### DISCUSSION TITLE
How to implement Typeahead without triggering loaders

### DISCUSSION THREAD

**Discussion by mhuretski** [remix-run/remix#6036] [Q&A] [answered]
I found a few hacks where you can trigger debounced form submission but there's no other way other than action to create custom endpoints in Remix. And action always trigger loader, so typeahead becomes extremely slow due to the fact app should load typeahead data and also ALL loaders in the route

Is there any way how action could be called without triggering any loader functions in the route?

**Comment by mhuretski** [comment, marked as answer]
I shouldn't have used action, I used loader in separate route file and fetched data directly with fetch API

**Reply by kiliman** [reply]
You can also use fetchers, which follow the Remix data handling workflow.

https://remix.run/docs/en/1.15.0/hooks/use-fetcher
